src/schema/user/user.resolver.ts

```diff
@@ -72,6 +72,40 @@ export default class UserResolvers {
     }
   }
 
+  /*
+    getUserByPhone query takes a phoneNumber as a parameter and returns the
+    corresponding user
+  */
+  @Query(() => User, {
+    nullable: true,
+    description:
+      'Takes an array of User ObjectIDs as a parameter and returns an array of corresponding users. If an empty array is passed, All the users are returned.',
+  })
+  async getUserByPhone(
+    @Arg('phoneNo')
+    phoneNo: string,
+    @Ctx() context: Context,
+  ): Promise<User | null> {
+    if (!HasPermissions(context, [PERMISSIONS.USER, PERMISSIONS.ADMIN])) {
+      throw new Error('Error: Unauthorized');
+    }
+
+    if (!phoneNo) {
+      throw new Error('Bad Request: Missing Parameter');
+    }
+
+    try {
+      const user = await UserModel.findOne({phoneNo});
```

Comment:
1. Make sure the field phone number is indexed on the database. It won't matter usually in small scale applications, but a good practice. 
2. It is a recommended practice to have two fields for phone number separately, country code and phone number separately to account for all possible variations seamlessly.